pkg/controller/v1alpha2/experiment/util/webhook_util.go

```diff
@@ -0,0 +1,88 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"fmt"
+
+	experimentsv1alpha2 "github.com/kubeflow/katib/pkg/api/operators/apis/experiment/v1alpha2"
+)
+
+func DefaultExperiment(instance *experimentsv1alpha2.Experiment) error {
+	instance.Default()
+	return nil
+}
+
+func ValidateExperiment(instance *experimentsv1alpha2.Experiment) error {
+	if err := validateObjective(instance.Spec.Objective); err != nil {
+		return nil
+	}
+
+	if err := validateAlgorithm(instance.Spec.Algorithm); err != nil {
+		return nil
+	}
+
+	if err := validateTrialTemplate(instance, instance.Spec.TrialTemplate); err != nil {
+		return nil
+	}
+
+	if len(instance.Spec.Parameters) == 0 && instance.Spec.NasConfig == nil {
+		return fmt.Errorf("spec.Parameters or spec.NasConfig must be specified.")
+	}
+
+	if len(instance.Spec.Parameters) > 0 && instance.Spec.NasConfig != nil {
+		return fmt.Errorf("Only one of spec.Parameters and spec.NasConfig can be specified.")
+	}
+
+	if len(instance.Spec.Parameters) > 0 {
+		if err := validateParameters(instance.Spec.Parameters); err != nil {
```

Comment:
Will we use `ValidateAlgorithmSetting` function to validate parameters inside Suggestion ? 
https://github.com/kubeflow/katib/blob/master/pkg/api/v1alpha2/api.proto#L170

Reply:
sure :)

Reply:
Good, thank you!